### ISSUE TITLE
Imlplement Q-learning algorithm

### ISSUE THREAD

**Issue by mkduer** [mkduer/reinforcement-learning-snake-game#10]
**hyperparameters** 
for producing proof of concept (in other words, keeping things simple to start)

* η, learning rate = 1
* γ, discount factor = 0 (no learning occurs, this should eventually be (0,1), 0.9 may be a good one to eventually use)
* r, reward = 2 for eating mouse (later on, penalties could be added like -1 for collision, 1 for surviving)
* ε, epsilon randomness is not used to start

Simple q-learning algorithm (with η = 1):
**Q(s, a) = (Q(s, a)) + (Q(s, a) + γ [maxa'(Q(s', a')))] - (Q(s, a)))**

Epsilon-Greedy q-learning algorithm:
Q(s, a) = (1 - ε)(Q(s, a)) + η (Q(s, a) + γ [maxa'(Q(s', a')))] - (Q(s, a)))
